components/plurals/src/rules/mod.rs

```diff
@@ -143,13 +144,5 @@
 //! [`Condition`]: super::rules::ast::Condition
 //! [`Sample`]: super::rules::ast::Samples
 //! [`AST`]: super::rules::ast
-pub mod ast;
-pub(crate) mod lexer;
-pub(crate) mod parser;
-pub(crate) mod resolver;
-pub(crate) mod serializer;
-
-pub use lexer::Lexer;
-pub use parser::{parse, parse_condition};
-pub use resolver::test_condition;
-pub use serializer::serialize;
+pub mod reference;
```

Comment:
Question: Why are you making these two modules `pub`? I want to understand your mental model around the visibility of the types in this crate.

Reply:
I imagine people using this crate in one of two ways:

```rust
use icu::plurals::PluralRules;

let pr = PluralRules::try_new(...).expect(...);
pr.select(5);
```

or:

```rust
use icu::plurals::rules::reference::{parse, ast, serialize};

let ast = parse("i % 10 = 5..3").expect(...);
if inspect_ast_to_learn_something(&ast) {
  let new_ast = modify_ast(&ast);
  let string = serialize(ast).expect(...);
  write_to_file(string);
}
```

For the latter, I imagine we may want to put it behind a feature, but since it doesn't loop in any new dependencies, we can just let DCE cut it when not needed.

Does it sound acceptable?